image.py

```diff
@@ -0,0 +1,66 @@
+import boto3
+from botocore.exceptions import ClientError
+from boto3.s3.transfer import S3UploadFailedError
+from pathlib import Path
+from PIL import Image
+from settings import AWS_ACCESS_KEY, AWS_SECRET_KEY, BUCKET_NAME
+import requests
+import logging
+import tempfile
+import io
+
+logger = logging.getLogger(__name__)
+ROOT = Path(__file__).parent.resolve()
+
+
+class ImageManagement(object):
+
+    def __init__(self):
+
+        self.s3 = None
+        self.temp_dir = None
+        self.session = boto3.Session(
+            aws_access_key_id=AWS_ACCESS_KEY,
+            aws_secret_access_key=AWS_SECRET_KEY
+        )
+
+    def download(self, image_url, image_name):
+        # create temp directory to save a file in it
+        self.temp_dir = tempfile.TemporaryDirectory()
+        r = requests.get(image_url, stream=True)
+        if r.status_code == 200:
```

Comment:
You didn't add a way to handle if the download fails. you will need to implement a try again technique and then a stop if it fails after three times for example